test/test_integration.py

```diff
@@ -672,6 +672,68 @@ def socket_handler(listener):
         recv_event.set()
         self.tear_down()
 
+    def test_secure_proxy_connection(self):
+        self.set_up(secure=True, secure_auto_wrap_socket=False, proxy=True)
+
+        data = []
+        req_event = threading.Event()
+        recv_event = threading.Event()
+
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+
+            # Read the CONNECT reader
+            connect_data = b''
+            while not connect_data.endswith(b'\r\n\r\n'):
+                connect_data += sock.recv(65535)
+
+            sock.send(b'HTTP/1.0 200 Connection established\r\n\r\n')
+
+            # todo make this less ugly?
+            sock = self.server_thread._wrap_socket(sock)
```

Comment:
I think this is fine, TBH. Maybe promote `_wrap_socket` to `wrap_socket`, or even pull that `wrap_socket` method off the server thread and make it available as a test helper function instead.